bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/ConfigParser.java

```diff
@@ -120,49 +141,81 @@
         return fields;
     }
 
-    private static Object objectConvert(Object value, Class<?> type) {
+    /**
+     * Convert a value to a given type
+     *
+     * @param value input value or String representation of that value
+     * @param type desired target class
+     * @param defaultValue default value to be used if conversion fails or input value is null
+     * @return the converted value
+     */
+    public static <T> T valueAs(@Nullable Object value, Class<T> type, T defaultValue) {
```

Comment:
We can name it `valueAsElse()` or `valueAsOrElse()` to let the code have a better readability.

```suggestion
    public static <T> T valueAsElse(@Nullable Object value, Class<T> type, T defaultValue) {
```